common/media-type/src/main/java/module-info.java

```diff
@@ -18,7 +18,7 @@
  * Helidon Common Media Type.
  */
 module io.helidon.common.media.type {
-    requires io.helidon.common;
+    requires transitive io.helidon.common;
```

Comment:
Why is this a transitive dependency?
`transitive` should only be used if types from that module are used in public API of this module. As far as I can see, only `HelidonServiceLoader` is used - and that is not an API usage, so it should not be a `transitive` dependency.
Please make sure this is correctly done in all the modules in this PR.

Reply:
addressed